### ISSUE TITLE
There doesn't seem to be a way to execute tasks or targets for 'dotnet run'

### ISSUE THREAD

**Issue by rolfbjarne** [dotnet/sdk#18436]
### Description

For mobile platforms (iOS, tvOS), we need to execute tasks to figure out a few things before we can run an app.

So we implemented a target to run before the `Run` target:

https://github.com/xamarin/xamarin-macios/blob/6ce5f80cb7ef96000652fdd534c9176b793402e9/dotnet/targets/Xamarin.Shared.Sdk.targets#L859

This works fine for `dotnet build -t:Run`, but it doesn't not work for `dotnet run`, because from what I can see `dotnet run` will pick out the `RunCommand` and `RunArguments` properties once the project has been evaluated:

https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/dotnet/commands/dotnet-run/RunCommand.cs#L236-L263

instead of executing the `Run` target, which means that the `RunCommand` and `RunArguments` properties aren't set, because we set them inside a target:

https://github.com/xamarin/xamarin-macios/blob/6ce5f80cb7ef96000652fdd534c9176b793402e9/dotnet/targets/Xamarin.Shared.Sdk.targets#L846-L849

The end result is that with an iOS app executing `dotnet run` just doesn't work:

* Install .NET 6 Preview 6 on a Mac.
* Install the iOS workload: `sudo dotnet workload install microsoft-ios-sdk-full`
* Clone https://github.com/dotnet/maui-samples (hash: bb264d046139f6635ed066157c44b99010b8d1dd)
* Build & run HelloiOS

```shell
$ cd HelloiOS
$ dotnet build
$ dotnet run --no-build
A fatal error was encountered. The library 'libhostpolicy.dylib' required to execute the application was not found in '/Users/rolf/work/net6-mobile-samples/HelloiOS/bin/Debug/net6.0-ios/iossimulator-x64/'.
Failed to run as a self-contained app.
  - The application was run as a self-contained app because '/Users/rolf/work/net6-mobile-samples/HelloiOS/bin/Debug/net6.0-ios/iossimulator-x64/HelloiOS.runtimeconfig.json' did not specify a framework.
  - If this should be a framework-dependent app, specify the appropriate framework in '/Users/rolf/work/net6-mobile-samples/HelloiOS/bin/Debug/net6.0-ios/iossimulator-x64/HelloiOS.runtimeconfig.json'
```

**Comment by marcpopMSFT** [comment]
This is a reasonable request but deferring to investigations around multi-project startup. @KathleenDollard

**Comment by rolfbjarne** [comment]
Any updates on this?

**Comment by baronfel** [comment]
Going to close this one as a duplicate of #31253, even though this one is older, since #31252 has more discussion on it.
